core/src/main/java/org/opensearch/sql/planner/optimizer/rule/EvalPushDown.java

```diff
@@ -42,6 +51,38 @@ public class EvalPushDown<T extends LogicalPlan> implements Rule<T> {
                 return logicalEval;
               });
 
+  public static final Rule<LogicalSort> PUSH_DOWN_SORT =
+      match(sort(evalCapture()))
+          .apply(
+              (sort, logicalEval) -> {
```

Comment:
Looks this Lambda is too large. Please extract to a method or a class. And extract some portion to reduce the size of method for readability.